PostHog/Models/Survey.swift

```diff
@@ -0,0 +1,346 @@
+/// see: posthog-js/posthog-surveys-types.ts
+struct Survey: Decodable {
+    /// The unique identifier for the survey
+    let id: String
+    /// The name of the survey
+    let name: String
+    /// Type of the survey (e.g., "popover")
+    let type: SurveyType
+    /// The questions asked in the survey
+    let questions: [SurveyQuestion]
+    /// Multiple feature flag keys. Must all (AND) evaluate to true for the survey to be shown (optional)
+    let featureFlagKeys: [SurveyFeatureFlagKeyValue]?
+    /// Linked feature flag key. Must evaluate to true for the survey to be shown (optional)
+    let linkedFlagKey: String?
+    /// Targeting feature flag key. Must evaluate to true for the survey to be shown (optional)
+    let targetingFlagKey: String?
+    /// Internal targeting flag key. Must evaluate to true for the survey to be shown (optional)
+    let internalTargetingFlagKey: String?
+    /// Conditions for displaying the survey (optional)
+    let conditions: SurveyConditions?
+    /// Appearance settings for the survey (optional)
+    let appearance: SurveyAppearance?
+    /// The iteration number for the survey (optional)
+    let currentIteration: Int?
+    /// The start date for the current iteration of the survey (optional)
+    let currentIterationStartDate: Date?
+    /// Start date of the survey (optional)
+    let startDate: Date?
+    /// End date of the survey (optional)
+    let endDate: Date?
+}
+
+struct SurveyFeatureFlagKeyValue: Equatable, Decodable {
+    let key: String
+    let value: String?
+}
+
+// MARK: - Question Models
+
+/// Protocol defining common properties for all survey question types
+protocol SurveyQuestionProperties {
+    /// Question text
+    var question: String { get }
+    /// Additional description or instructions (optional)
+    var description: String? { get }
+    /// Content type of the description (e.g., "text", "html") (optional)
+    var descriptionContentType: SurveyTextContentType? { get }
+    /// Indicates if this question is optional (optional)
+    var optional: Bool? { get }
+    /// Text for the main CTA associated with this question (optional)
+    var buttonText: String? { get }
+    /// Original index of the question in the survey (optional)
+    var originalQuestionIndex: Int? { get }
+    /// Question branching logic if any (optional)
+    var branching: SurveyQuestionBranching? { get }
+}
+
+/// Represents different types of survey questions with their associated data
+enum SurveyQuestion: Decodable {
+    case basic(BasicSurveyQuestion)
+    case link(LinkSurveyQuestion)
+    case rating(RatingSurveyQuestion)
+    case multipleChoice(MultipleSurveyQuestion)
+    case singleChoice(MultipleSurveyQuestion)
+
+    init(from decoder: Decoder) throws {
+        let container = try decoder.container(keyedBy: CodingKeys.self)
+        let type = try container.decode(SurveyQuestionType.self, forKey: .type)
```

Comment:
when the code throws we need to be sure to swallow exception and log out and not crash peoples apps